docker/functions.go

```diff
@@ -34,3 +34,24 @@ func GetContainerByName(client *dockerclient.Client, name string) (*dockerclient
 
 	return &containers[0], nil
 }
+
+// GetContainerByID looks up the hosts containers with the specified Id and
+// returns it, or an error.
+func GetContainerByID(client *dockerclient.Client, id string) (*dockerclient.APIContainers, error) {
+	containers, err := client.ListContainers(dockerclient.ListContainersOptions{All: true, Size: false})
```

Comment:
I wonder why we get all the container and then filter them out ? `ListContainersOptions` and the API has a way to use filter, and one of them is `id`, so we would let `docker` do the hard work here. Something like :

```go
containers, err := client.ListContainers(dockerclient.ListContainersOptions{
    All: true, 
    Filter: map[string][]string{
        "id":id,
    }, 
    Size: false,
})
```

Then, the only thing is to check the length :wink: